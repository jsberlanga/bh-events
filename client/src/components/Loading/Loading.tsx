import styled from 'styled-components';

const StyledLoader = styled.div`
  display: inline-block;
  width: 40px;
  height: 40px;

  div {
    box-sizing: border-box;
    display: block;
    position: absolute;
    width: 40px;
    height: 40px;
    border: 2px solid#222831;
    border-radius: 50%;
    animation: lds-ring 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
    border-color: #222831 transparent transparent transparent;

    &::nth-of-type(1) {
      animation-delay: -0.45s;
    }
    &::nth-of-type(2) {
      animation-delay: -0.3s;
    }
    &::nth-of-type(3) {
      animation-delay: -0.15s;
    }
  }

  @keyframes lds-ring {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`;

const Loading = () => (
  <StyledLoader className="lds-ring">
    <div></div>
    <div></div>
    <div></div>
    <div></div>
  </StyledLoader>
);

export default Loading;
