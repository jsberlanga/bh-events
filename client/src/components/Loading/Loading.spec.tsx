import { render } from '@testing-library/react';
import Loading from './Loading';

describe('CreateEvent', () => {
  it('should render with styles correctly', () => {
    const { container } = render(<Loading />);

    expect(container).toMatchSnapshot();
  });
});
