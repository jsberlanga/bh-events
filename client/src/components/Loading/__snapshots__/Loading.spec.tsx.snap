// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateEvent should render with styles correctly 1`] = `
.c0 {
  display: inline-block;
  width: 40px;
  height: 40px;
}

.c0 div {
  box-sizing: border-box;
  display: block;
  position: absolute;
  width: 40px;
  height: 40px;
  border: 2px solid#222831;
  border-radius: 50%;
  -webkit-animation: lds-ring 1.2s cubic-bezier(0.5,0,0.5,1) infinite;
  animation: lds-ring 1.2s cubic-bezier(0.5,0,0.5,1) infinite;
  border-color: #222831 transparent transparent transparent;
}

.c0 div::nth-of-type(1) {
  -webkit-animation-delay: -0.45s;
  animation-delay: -0.45s;
}

.c0 div::nth-of-type(2) {
  -webkit-animation-delay: -0.3s;
  animation-delay: -0.3s;
}

.c0 div::nth-of-type(3) {
  -webkit-animation-delay: -0.15s;
  animation-delay: -0.15s;
}

<div>
  <div
    class="c0 lds-ring"
  >
    <div />
    <div />
    <div />
    <div />
  </div>
</div>
`;
